src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import type { HTMLProps, ReactNode } from "react";
+
+import Field from "../Field";
+
+import type { PropsWithSpread } from "types";
+
+export type Props = PropsWithSpread<
+  {
+    /**
+     * The label name for the switch
+     */
+    label: ReactNode;
+    /**
+     * Whether the switch is disabled or not
+     */
+    disabled?: boolean;
+  },
+  HTMLProps<HTMLInputElement>
+>;
+
+export const Switch = ({
+  label,
+  disabled = false,
+  ...inputProps
+}: Props): JSX.Element => {
+  return (
+    <Field>
```

Comment:
I don't think we need to wrap it in Field component. It adds 2 additional wrapping divs for "form group" and "form validation".

This should just be a switch component without any additional form markup. So it's enough if `<label>` is root of this element.

Reply:
Thanks, updated! @bartaz